packages/mui-material/src/styles/ThemeProvider.tsx

```diff
@@ -26,6 +26,14 @@ type ThemeProviderCssVariablesProps = CssThemeVariables extends { enabled: true
        * @default false
        */
       disableStyleSheetGeneration?: boolean;
+      /**
+       * If `true`, theme values will not change when the mode/color-scheme changes.
+       *
+       * This is useful for optimizing rendering performance. However, need to ensure that all components
+       * are using CSS variables or static values that does not differ between modes to prevent SSR hydration mismatch.
+       * @default false
+       */
+      freezeThemeValues?: boolean;
```

Comment:
Another name is `freezeThemeBetweenModes`, which one do you prefer? @aarongarciah @DiegoAndai

Reply:
No strong opinion. If we go with `freezeThemeValues`, maybe we can simplify it further to be `freezeTheme`?

Reply:
Well, `freezeTheme` is probably confusing because a theme also contains component style overrides and such.